src/components/UserDashboard.jsx

```diff
@@ -0,0 +1,42 @@
+import { useContext } from "react"
+import { Link } from "react-router-dom"
+import UserContext from "../context/UserInfo"
+function UserDashboard() {
+  const { user, setUser, setState } = useContext(UserContext)
```

Comment:
This could potentially be refactored into a custom React hook, so you don't have to import `UserContext` and `useContext` every time.

Instead it would allow you to import the hook by itself.

```JSX
// this
import { useContext } from "react"
import UserContext from "../context/UserInfo"

// vs this
import useCustomContext from "../context/UserInfo"
// i dunno, maybe come up with a better name

```